File: packages/typia/src/factories/internal/metadata/iterate_metadata_comment_tags.ts
import { MetadataObjectType } from "../../../schemas/metadata/MetadataObjectType";

import { MetadataCommentTagFactory } from "../../MetadataCommentTagFactory";
import { MetadataFactory } from "../../MetadataFactory";

export const iterate_metadata_comment_tags = (props: {
  errors: MetadataFactory.IError[];
  object: MetadataObjectType;
}) => {
  if (props.object.tagged_ === true) return;
  props.object.tagged_ = true;

  for (const property of props.object.properties) {
    MetadataCommentTagFactory.analyze({
      errors: props.errors,
      metadata: property.value,
      tags: property.jsDocTags,
      explore: {
        top: false,
        object: props.object,
        property: property.key.isSoleLiteral()
          ? property.key.getSoleLiteral()!
          : {},
        parameter: null,
        nested: null,
        aliased: false,
        escaped: false,
        output: false,
      },
    });
  }
};

File: packages/core/src/factories/internal/metadata/iterate_metadata_comment_tags.go
package metadata

func IterateMetadataCommentTags(props struct {
	Errors *MetadataErrors
	Object *MetadataObjectType
}) {
	if props.Object == nil || props.Object.Tagged {
		return
	}
	props.Object.Tagged = true
	for _, property := range props.Object.Properties {
		AnalyzeCommentTags(struct {
			Errors   *MetadataErrors
			Metadata *MetadataSchema
			Tags     []JSDocTag
			Explore  MetadataExplore
		}{
			Errors: props.Errors, Metadata: property.Value, Tags: property.JSDocTags,
			Explore: MetadataExplore{Object: props.Object, Property: property.Key.GetSoleLiteral()},
		})
	}
}
